redbot/cogs/downloader/downloader.py

```diff
@@ -402,7 +402,7 @@ def __init__(self, bot: Red):
             if can_react:
                 with contextlib.suppress(discord.Forbidden):
                     await query.clear_reactions()
-
+            cognames &= set(ctx.bot.cogs.keys())  # only reload loaded cogs
```

Comment:
This needs to be `await ctx.bot.db.packages()` rather than `ctx.bot.cogs.keys()`, and also doing this here means `cognames` as it is passed to `[p]reload` can be empty. Perhaps it is best to create this set before prompting for a reload and exit early if it's empty.

Reply:
Using `ctx.bot.extensions.keys()` instead. I initially went with that approach but we can skip a co-routine here, and this can be incorrect if the bot is loaded with the `--no-cogs` arg (as you noted in discord later)